gensim/models/phrases.py

```diff
@@ -64,8 +64,8 @@
 import warnings
 from collections import defaultdict
 import itertools as it
-from functools import partial
 from math import log
+from inspect import getargspec
```

Comment:
Import not used?

Reply:
Used in line 188 (in the commit your comments are on) to check for the proper parameters in the pluggable scoring function.

Reply:
Thanks, I see it now. What is that check for though? Python is duck-typed by convention, so "type checks" are best postponed until truly needed (something breaks).

Reply:
Mostly to save the stress that would result from improperly specifying a scoring function when initializing the phrases object. I know Python will do the type checking when the scoring function is called, but that won't happen until export_phrases or __getitem__ is called. The "normal" workflow for the Phrases object is to just specify sentences on load, or to use add_vocab. Only after that does the scoring function get called.

I could easily see a user specifying a bad scoring method and then making the vocab dictionary from their large corpus. Only after significant time extracting vocab from a corpus do they then discover that something is wrong with how they specified scoring. At this point you could manually specify a correct scoring function, but that requires you to set it directly. Users also wouldn't have an easy bailout in the form of use one of the scorer string settings, since those are only checked when the Phrases object is created--the user would have to figure out how to specify those built in scorers which would mean opening up the code. This seems a bit user unfriendly, I feel it is friendlier to just do the type checking on initialization even if it is less Pythonic.

This could be fixed with a set_scorer method that takes the string or function input, but that seems a bit more awkward than just doing this type check.

There's also an issue with wanting to raise an informative exception when the scoring function is called in __getitem__ or export_phrases and the types don't match, but that means adding a try/except into the main scoring loop and that seems awkward as well. I think its better to just do that try/except once when the object is initialized.

But I defer to your judgement on this--what do you think is best?

Reply:
Thanks, I see your argument (that checking early a little more convenient).

I'm not sure if it's worth it, but don't care much either way. I'll defer to @menshikh-iv :)